Majasdarbs/07-majasdarbs/OlegsKozlovs/app.js

```diff
@@ -0,0 +1,120 @@
+// TODO app (dienas planotaju)
+// Nokopejiet šo mapi savas majas darba mape
+// Aplikacijas darbas principi var apskatit video appDemo.mp4
+
+/* Mums nepieciešams izviedot nelielu todo aplikaciju ar iespeju pievienot izdzest un atziment izdarito notikumu
+    Mums nepieciešams 4 funkcijas
+        addTask() - pievienot notikumu
+            1 - izveleties inputa vertibu izmantojot selector.value;
+            2 - izveido object priekš notikums 
+                const task = {
+                    textTask,
+                    done: false
+                }
+            3 - izsaukt funkciju saveToLocalStorage
+
+        saveToLocalStorage() - saglaba notikumu
+            1 - jaizmanto JSON.stringify
+            2 - jaizmanto atslegst vards 'taskList'
+
+        renderTask() - izvada sarakstu notikumu
+            1 - jaizmanto innerHTML
+            2 - jaizmanto vienu no cikliem lai iziet cauri visam localstorage ierakstiem
+            3 - šis html palidzes izvadit datus
+            return `<li data-index='${i}'>
+                        <div class="">
+                            ${data.textTask}<span class="remove">❌</span>
+                        </div>
+                    </li>`;
+            4 - Gadijuma ja elements ir apziments ka izdarits mums nepiecišams pievienot klassi .done ../style.css
+                Lidz ar to nepieciešams izveidot parbaudi if else lai parbaudit test.done === 'true'
+
+        toggleDone() - atzimet ka izdarito
+            Mums nepiecišams pievinot 2 eventListener
+                addEventListener('submit', addTask);   ----> nostradas kad mes nospiedam pievinot pogu un izsaucam funkciju addTask
+                addEventListener('click', toggleDone); ----> nostradas kad mes nospiedam uz elementu saraksta un izsaucam funkciju toggleDone
+
+                funkcija toogleDone dara divas darbibas
+                gadijuma ja bija nospiesta izdžešanas poga, mes izdesam elementu no localStorage izsaucam renderTask funkciju lai atjaunto sarakstu
+                gadijuma ja bija nospiest elements mes nomainam elementam done vertibu done: false --> done: true un izsaucam renderTask funkciju lai atjauno sarakstu
+*/
+// addEventListener click
+// addEventListener submit
+
+const taskListElement = document.getElementById("task-list");
+const addButton = document.getElementById("add-button");
+const inputField = document.getElementById("input");
+
+const taskList = JSON.parse(localStorage.getItem("task_list")) || [];
+
+const saveToLocalStorage = (key, value) => {
+  localStorage.setItem(key, JSON.stringify(value));
+};
+
+const isInTaskList = (elem) => {
+  for (let i = 0; i < taskList.length; i++) {
+    if (taskList[i].value === elem) {
+      return true;
+    }
+  }
+  return false;
+}
+
+const isValidInput = (input) => {
+  if (input === null || input.trim().length === 0) {
```

Comment:
You can remove this validation and make field required in html https://www.w3schools.com/tags/att_input_required.asp

Reply:
Thanks for comment. I have tried to do that but for some reason "required" option in HTML doesn't work for me. I have done the following: <input type="text" name="task" placeholder="New Task Item" id="input" required />
Reviewing potential problems with this option in Google, I can't find the root of the issue. Could you please advise?